design-patterns/src/decorators/ShipmentDecorator.ts

```diff
@@ -0,0 +1,24 @@
+import { IShipment, Marks } from "../types";
+
+type Constructor<T = {}> = new (...args: any[]) => T;
+type Shippable = Constructor<IShipment>;
+
+export function ShipmentDecorator<T extends Shippable>(constructor: T) {
```

Comment:
it's not ok to give any postfix for naming as *Decorator

better something as `@Shippable` or `@ShipWithMarks`